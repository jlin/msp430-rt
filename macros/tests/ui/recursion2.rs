#![no_main]

use msp430_rt_macros::{entry, interrupt};

#[entry]
fn main(cs: CriticalSection) -> ! {
    main()
}

#[interrupt]
fn DefaultHandler(cs: CriticalSection) -> ! {
    DefaultHandler()
}
