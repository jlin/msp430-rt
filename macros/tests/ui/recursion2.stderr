error[E0423]: expected function, found built-in attribute `main`
 --> tests/ui/recursion2.rs:7:5
  |
7 |     main()
  |     ^^^^ not a function

error[E0425]: cannot find function, tuple struct or tuple variant `DefaultHandler` in this scope
  --> tests/ui/recursion2.rs:12:5
   |
12 |     DefaultHandler()
   |     ^^^^^^^^^^^^^^ not found in this scope

error[E0658]: msp430-interrupt ABI is experimental and subject to change
  --> tests/ui/recursion2.rs:10:1
   |
10 | #[interrupt]
   | ^^^^^^^^^^^^
   |
   = note: see issue #38487 <https://github.com/rust-lang/rust/issues/38487> for more information
   = help: add `#![feature(abi_msp430_interrupt)]` to the crate attributes to enable
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0570]: `"msp430-interrupt"` is not a supported ABI for the current target
  --> tests/ui/recursion2.rs:10:1
   |
10 | #[interrupt]
   | ^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)
